test/connectivity/NetworkController/Program.cs

```diff
@@ -0,0 +1,105 @@
+            var tcs = new TaskCompletionSource<bool>();
+            cts.Token.Register(s => ((TaskCompletionSource<bool>)s).SetResult(true), tcs);
+
+            switch (Settings.Current.Mode)
+            {
+                case Mode.Offline:
+                    StartOfflineMode();
+                    break;
+                case Mode.Satellite:
+                    StartSatelliteMode();
+                    break;
+            }
+
+            await tcs.Task;
+            return 0;
+        }
+
+        static void StartSatelliteMode() => throw new NotImplementedException();
+
+        static void StartOfflineMode()
+        {
+            var delay = Settings.Current.StartAfter;
+            var networkInterfaceName = GetDockerInterfaceName();
+            networkInterfaceName.Match(
+                name =>
+                {
+                    delay = StartOfflineScheduledTask(delay, new NetworkInterfaceOfflineController(name));
+                    StartOfflineScheduledTask(delay, new FirewallOfflineController(name));
+                    return Task.CompletedTask;
+                },
+                () =>
+                {
+                    Log.LogError($"No network interface found for docker network {Settings.Current.NetworkId}");
+                    return Task.CompletedTask;
+                }
+            );
+        }
+
+        static TimeSpan StartOfflineScheduledTask(TimeSpan delay, IOfflineController offlineController)
+        {
+            foreach (var item in Settings.Current.Frequency)
+            {
+                Log.LogInformation(
+                    $"Schedule periodic task with {offlineController.Description} to start after {delay} Offline frequency {item.OfflineFrequency} Online frequency {item.OnlineFrequency} Run times {item.RunsCount}");
+                new ScheduledTask(
+                    async cs =>
+                    {
+                        await offlineController.SetOffline(cs);
+                        await Task.Delay(item.OfflineFrequency, cs);
+                        await offlineController.SetOnline(cs);
+                    },
+                    delay,
+                    item.OnlineFrequency,
+                    item.RunsCount,
+                    Log,
+                    "enable/disable");
+
+                delay += (item.OfflineFrequency + item.OnlineFrequency) * item.RunsCount;
+            }
+
+            return delay;
+        }
+
+        static Option<string> GetDockerInterfaceName()
+        {
+            var client = new DockerClientConfiguration(new Uri(Settings.Current.DockerUri)).CreateClient();
+            var networkId = client.Networks.InspectNetworkAsync(Settings.Current.NetworkId).Result.ID;
+
+            Log.LogInformation($"{Settings.Current.NetworkId} network has id {networkId}");
+
+            foreach (var item in NetworkInterface.GetAllNetworkInterfaces())
+                if (item.Name.Contains(networkId.Substring(0, 12)))
```

Comment:
Magic? :) Can you add some comments?

Reply:
I added constants and a comment to be more clear

---
In reply to: [353015981](https://github.com/Azure/iotedge/pull/2040#discussion_r353015981) [](ancestors = 353015981)